GCR support only works/is useful with X11 support.

Reported upstream: https://bugs.launchpad.net/midori/+bug/1515985

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura midori-0.5.11.orig/CMakeLists.txt midori-0.5.11/CMakeLists.txt
--- midori-0.5.11.orig/CMakeLists.txt	2015-11-12 13:41:02.045898814 -0300
+++ midori-0.5.11/CMakeLists.txt	2015-11-12 14:00:09.765545018 -0300
@@ -175,18 +175,14 @@
     set(PKGS ${PKGS} zeitgeist-2.0)
 endif()
 
-if (WIN32)
-    add_definitions("-DGCR_VERSION=\"No\"")
-else ()
-    if (USE_GTK3)
-        pkg_check_modules(GCR REQUIRED gcr-3>=2.32)
-    else ()
-        pkg_check_modules(GCR REQUIRED gcr-base-3>=2.32)
-    endif ()
+if (USE_GTK3 AND X11)
+    pkg_check_modules(GCR REQUIRED gcr-3>=2.32)
     add_definitions("-DGCR_VERSION=\"${GCR_VERSION}\"")
     add_definitions("-DHAVE_GCR")
     set(OPTS_INCLUDE_DIRS ${OPTS_INCLUDE_DIRS} ${GCR_INCLUDE_DIRS})
     set(OPTS_LIBRARIES ${OPTS_LIBRARIES} ${GCR_LIBRARIES})
+else ()
+    add_definitions("-DGCR_VERSION=\"No\"")
 endif ()
 
 if (HALF_BRO_INCOM_WEBKIT2)
